### ISSUE TITLE
Recalculate exp after test case change

### ISSUE THREAD

**Issue by trangmei** [Coursemology/coursemology2#3110]
After https://github.com/Coursemology/coursemology2/pull/3087
1. creators of autograded assessments can specify which types of test cases will be counted towards grade; and
2. assessment creator can allow students to finalise autograded assessments at any point, i.e. they do not have to get **all** questions to pass pubic and private test cases before the 'Finalise Submission' button appears.
For assessments with such settings (use case: CS1010S autograded missions), when creators change the collection of graded test cases, by changing either the assessment's booleans or adding/removing test cases, respective answers' grades are re-evaluated. However, the exp given are not dynamically calculated, hence it remains unchanged.
We need to revise the exp upon such changes, i.e. change the submission's points_awarded in DB.

#TODO:
- [x] Create submission event `resubmit_programming` which unsubmit published submissions, unsubmit all their current answers to programming questions, change submissions' state from published to submitted.
- [x] Pass a toggle to submission autograding service to only grade ungraded_answers
Basically switching off https://github.com/Coursemology/coursemology2/pull/2566 for this special use case (test case add/removal).
- [x] Write test to cover `resubmit_programming`
